### ISSUE TITLE
[2021] Intrinsic Dimensionality Explains the Effectiveness of Language Model Fine-Tuning

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#152]
## 0. 論文

| タイトル | Intrinsic Dimensionality Explains the Effectiveness of Language Model Fine-Tuning |
|:-------|:--|
| リンク  | https://arxiv.org/abs/2012.13255 |
| 著者    | Armen Aghajanyan, Luke Zettlemoyer, Sonal Gupta (Facebook) |
| 投稿日付 | ACL2021 |

## 1. どんなもの？

事前学習済みの言語モデルに対する fine-tuning の挙動を、intrinsic dimension (潜在次元) の観点から分析した研究。事前学習済みモデルが実は低い intrinsic dimension を持っており、下位タスクの fine-tuning に低次元の空間で対応できることを実験的に確認。例えば MRPC タスクに RoBERTa で、たった200個のパラメータで全パラメータ時スコアの90%の性能を達成した。
加えて、大きい事前学習モデルの方が低い intrinsic dimension を持っていること、合わせて低い intrinsic dimension が汎化性能とも相関することも確認した。
また、compression ベースの汎化境界と同じ方法で intrinsic dimension ベースの汎化境界を導出している。

## 2. 先行研究と比べてどこがすごい？

- intrinsic dimension に着目して、まだ解明されてない点の多い pre-trained モデルの fine-tuning 時の挙動について分析している。

## 3. 技術や手法のキモはどこ？

- intrinsic dimension
    - モデルのD個のパラメータを θ^D を、適当なDより小さい d個のパラメータ θ^d を用いて、次のように決める
        - ![スクリーンショット 2021-09-21 22 18 09](https://user-images.githubusercontent.com/1255485/134177630-7dd4ef0f-3ddc-4bcb-8aba-214a13070019.png)
    - 学習時は、 θ^d だけを学習し、P (ランダム行列などを使う) で元のパラメータ空間に写像する
    - 元の精度の 90% の精度が出る最小の値 d を探索し、それを d_{90} として intrinsic dimension として定義する
- 工夫と改良
    - P はランダム行列と書かれているが、実際は巨大な事前学習モデルの空間への写像を作る必要からランダム行列ではメモリが足らなくなるため、fastfood transform を使用する
    - パラメータのうちの1つをレイヤーごとの scaling に当てる Structure Aware Intrinsic Dimension も試しており、こっちの方が通常よりちょっと良い
        - ![スクリーンショット 2021-09-21 22 26 24](https://user-images.githubusercontent.com/1255485/134179061-197b0b75-7359-461e-9e17-103937896f2c.png)

## 4. どうやって有効だと検証した？

- 実験内容
    - intrinsic dimension による学習を d を 10~10000 で変えながら行い、その精度をオリジナルと比較する
- ベンチマークデータセット
    - MRPC と QQP (データ数小と大)
- モデル
    - BERT と RoBERTa の base と large それぞれ
- 結果
    - 横軸が intrinsic dimension d, 縦軸が metric で、下図は DID を用いた場合の結果。
    - ![スクリーンショット 2021-09-21 0 15 39](https://user-images.githubusercontent.com/1255485/134027479-37efb5fd-ac0c-4101-ae4b-77822e1b4c87.png)
- その他、詳細割愛するが次を実験で確認した
    - intrinsic dimension は pre-training が進むほど小さくなる
    - intrinsic dimension は学習方法よりもパラメータ数に依存し、パラメータ数が大きいほど小さくなる
    - intrinsic dimension は汎化性能とも相関があり、より小さい方が汎化性能が良い

## 5. 議論はある？

- intrinsic dimension の最適化と SGD の理論的な関連は future work としている
